### DISCUSSION TITLE
Pattern for custom get_param_as_...

### DISCUSSION THREAD

**Discussion by davetapley** [falconry/falcon#2334] [Q&A] [answered]
What's the best way to implement my own `get_params_as_...` on `Request`
(see e.g. [`get_param_as_datetime`](https://falcon.readthedocs.io/en/stable/api/request_and_response_asgi.html#falcon.asgi.Request.get_param_as_datetime))?

Ideally so I can reuse it in multiple resources.

For context, I need a Pandas [`Timestamp`](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Timestamp.html).

**Comment by vytas7** [comment, marked as answer]
Hi @davetapley!
There are many ways to achieve this depending on your usage patterns. If you do like the default `get_param_as_*()`, you could maybe simply subclass `Request`, and pass the new class to your [`App`](https://falcon.readthedocs.io/en/stable/api/app.html#wsgi-app)'s initializer?

**Reply by davetapley** [reply]
Ah I didn't see `request_type`, yep that'll work great as a mixin 🙏🏻
